src/main/java/com/commercetools/sync/commons/BaseSync.java

```diff
@@ -22,44 +26,54 @@ protected BaseSync(@Nonnull final U statistics, @Nonnull final V syncOptions) {
 
 
     /**
      * Given a list of resource (e.g. categories, products, etc..) drafts. This method compares each new resource in
      * this list with it's corresponding old resource in a given CTP project, and in turn it either issues update
      * actions on the existing resource if it exists or create it if it doesn't.
      *
-     * <p>The time before and after the actual sync process starts is recorded in the {@link BaseSyncStatistics}
-     * container so that the total processing time is computed in the statistics.
-     *
      * @param resourceDrafts the list of new resources as drafts.
      * @return an instance of {@link CompletionStage&lt;U&gt;} which contains as a result an instance of {@link U} which
-     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} instance attribute of
-     *      {@link this} {@link BaseSync}.
+     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} of this sync process.
      */
     public CompletionStage<U> sync(@Nonnull final List<T> resourceDrafts) {
-        statistics.startTimer();
-        return process(resourceDrafts).thenApply(resultingStatistics -> {
-            resultingStatistics.calculateProcessingTime();
+        final long startTime = System.currentTimeMillis();
+        return process(resourceDrafts).thenApply(resultingStatisticsBuilder -> {
+            resultingStatisticsBuilder.setProcessingTimeInMillis(System.currentTimeMillis() - startTime);
+            final U resultingStatistics = resultingStatisticsBuilder.build();
+            updateTotalStatistics(resultingStatistics);
             return resultingStatistics;
         });
     }
 
 
     /**
-     * Returns an instance of type U which is a subclass of {@link BaseSyncStatistics} containing all the stats of the
-     * sync process; which includes a report message, the total number of update, created, failed, processed resources
-     * and the processing time of the sync in different time units and in a human readable format.
+     * Returns an instance of type U which is a subclass of {@link BaseSyncStatistics} containing the total stats of
+     * all sync processes performed by {@code this} sync instance.
      *
-     * @return a statistics object for the sync process.
+     * @return a statistics object for all sync processes performed by {@code this} sync instance.
      */
     @Nonnull
     public U getStatistics() {
-        return statistics;
+        synchronized (totalStatisticsLock) {
+            return totalStatisticsBuilder.build();
+        }
+    }
+
+    /**
+     * Join given statistics to total stats of all sync processes performed by {@code this} sync instance.
+     *
+     * @param statisticsToJoin statistics of sync that should be joined to total stats
+     */
+    private void updateTotalStatistics(final U statisticsToJoin) {
+        synchronized (totalStatisticsLock) {
```

Comment:
To avoid inconsistent state. 
When one thread is updating fields the other could call `getStatistics()` and obtain the statistics object with partially updated counters what would be wrong.
That's why I decided to use `lock` which prevent from concurrent access to `totalStatisticsBuilder`.